code/fireball/fireballs.cpp

```diff
@@ -31,13 +31,15 @@ int Knossos_warp_ani_used;
 
 #define MAX_WARP_LOD	0
 
-fireball Fireballs[MAX_FIREBALLS];
+constexpr int INTITIAL_FIREBALL_SIZE = 256;
+
+SCP_vector<fireball> Fireball_vec(INTITIAL_FIREBALL_SIZE);
+SCP_vector<int> Unused_fireball_indices(INTITIAL_FIREBALL_SIZE);
```

Comment:
Careful with the vector constructor call. It _adds_ that many elements and does not only reserve as much. Since you already have the relevant code to reserve the required size below, you can just use the default constructor here.